### PR TITLE
192 like posts

### LINKED ISSUE #192: Like posts
**Current state**
There is no back-end or front-end for liking posts within the application

**Expected state**
There is a back-end or front-end for liking posts within the application

**Todo**
Backend
- [x] Creating table for likes
- [x] Creating update likes functionality in post DAO

Frontend
- [x] Implementing styling
- [x] Implementing frontend like functionality

**Necessities**
n/a

**Branch**
192-like-posts

### PR DESCRIPTION
**Pull request by jerohero**
closes #192

 **What is it supposed to do**
Like posts (or undo likes)

 **How can we test your branch**
Try it out I suppose

 **What broweser have been checked**
- [X] Chrome
- [X] Firefox
- [ ] Opera
- [X] Internet Explorer (The whole thing won't even load 😃)
- [X] Edge
- [ ] Safari

### CODE REVIEW

File: backend/app/controllers/postController.ts [hunk 1]
```diff
@@ -16,6 +17,9 @@ class PostController implements ControllerBase{
         this.router.get(this.post, this.postService.getPosts)
         this.router.get(this.post + '/:path', this.postService.getPostByPath)
         this.router.post(this.post, this.postService.createPost)
+        this.router.post(this.post + '/like/:path', isAuth, this.postService.likePost)
+        this.router.delete(this.post + '/like/:path', isAuth, this.postService.unlikePost)
```
Comment (AjobK) [inline backend/app/controllers/postController.ts]: This is a dummy, I dislike this :-(

File: backend/app/entity/post_like.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+@Entity('post_like')
+export class PostLike extends BaseEntity {
+    @PrimaryGeneratedColumn()
+    id: number;
+
+    @ManyToOne(() => User, user => user.id)
+    @JoinColumn({ name: 'user_id', referencedColumnName: 'id' })
+    user: User
+
```
Comment (AjobK) [inline backend/app/entity/post_like.ts:13]: Not nice (Test) 👎

**Review by AjobK — CHANGES_REQUESTED** [review]
Some things were wrong 😞

File: backend/app/entity/post_like.ts (hunk 2 above)
Comment (jerohero) [reply backend/app/entity/post_like.ts:13]: mean 😡

**Comment by AjobK** [comment]
@jerohero Has conflicts that need to be resolved. Most likely due to usage of 'user' which has been renamed to 'profile'

File: frontend/src/components/postLike/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        super(props)
+        this.toggleLike = this.props.toggleLike.bind(this)
+    }
+
+    postLike = () => {
+        const path = window.location.pathname.split('/').filter(i => i != '').pop()
+
+        Axios.post(`/post/like/${path}`, {}, {withCredentials: true})
+            .then(res => {
+                console.log(res.data.message)
```
Comment (AjobK) [inline frontend/src/components/postLike/index.js]: Console.log

File: frontend/src/components/postLike/index.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            })
+
+    }
+
+    postUnlike = () => {
+        const path = window.location.pathname.split('/').filter(i => i != '').pop()
+
+        Axios.delete(`/post/like/${path}`, {withCredentials: true})
+            .then(res => {
+                console.log(res.data.message)
```
Comment (AjobK) [inline frontend/src/components/postLike/index.js]: Console.log

File: frontend/src/components/postLike/index.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                this.toggleLike()
+            })
+            .catch(err => {
+                console.log(err)
```
Comment (AjobK) [inline frontend/src/components/postLike/index.js]: Console.log

File: frontend/src/pages/post/index.js [hunk 6]
```diff
@@ -31,31 +36,57 @@ class Post extends App {
...
+                    user_liked: res.data.likes.user_liked
+                }
             }
 
             this.setState({
                 post: this.post
             })
         })
+        .catch(err => {
+            console.log('Error occurred while fetching post')
```
Comment (AjobK) [inline frontend/src/pages/post/index.js]: Console.log

File: frontend/src/pages/post/index.js [hunk 7]
```diff
@@ -31,31 +36,57 @@ class Post extends App {
...
+        })
+    }
+
+    toggleLike = () => {
+        // Toggles liked state for all like components
+        console.log('LIKE TOGGLED')
```
Comment (AjobK) [inline frontend/src/pages/post/index.js]: Console.log

File: frontend/src/components/postLike/index.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        const path = window.location.pathname.split('/').filter(i => i != '').pop()
+
+        Axios.post(`/post/like/${path}`, {}, {withCredentials: true})
+            .then(res => {
+                console.log(res.data.message)
+                this.toggleLike()
+            })
+            .catch(err => {
+                if (err.response.status === 401) {
+                    alert('Not signed in!')
```
Comment (AjobK) [inline frontend/src/components/postLike/index.js]: Replace with redirect to login screen

File: frontend/src/pages/post/index.js [hunk 9]
```diff
@@ -31,31 +36,57 @@ class Post extends App {
...
-                content: json.content,
-                description: json.description,
-                path: path
+                title: res.data.post.title,
+                content: res.data.post.content,
+                description: res.data.post.description,
+                path: path,
+                likes: {
+                    amount: res.data.likes.amount,
+                    user_liked: res.data.likes.user_liked
```
Comment (AjobK) [inline frontend/src/pages/post/index.js]: Camelcase if possible please...

File: frontend/src/pages/post/index.js [hunk 10]
```diff
@@ -2,20 +2,25 @@ import React from 'react'
...
-            description: null,
-            content: null,
-            path: null
+            title: 'Loading..',
+            description: 'Loading..',
+            content: 'Loading..',
+            path: 'Loading..',
+            likes: {
+                amount: 0,
+                user_liked: false
```
Comment (AjobK) [inline frontend/src/pages/post/index.js]: If possible camelcase

**Review by AjobK — CHANGES_REQUESTED** [review]
FEATURE MISSING?: Please redirect to login when user tries to like when logged out

Infinite request when removing like twice

File: frontend/src/pages/post/index.js [hunk 11]
```diff
@@ -66,17 +59,25 @@ class Post extends App {
...
-                    description: json.description,
-                    path: path
+                    title: res.data.title,
+                    content: res.data.content,
+                    description: res.data.description,
+                    path: path,
+                    likes: {
+                        amount: res.data.likes.amount,
+                        userLiked: res.data.likes.userLiked
+                    }
```
Comment (jerohero) [inline frontend/src/pages/post/index.js]: What is all this for? (from merge with develop)

File: backend/app/entity/post_like.ts [hunk 12]
```diff
@@ -0,0 +1,21 @@
+import { BaseEntity, Column, Entity, JoinColumn, ManyToOne, OneToMany, ManyToMany, OneToOne, PrimaryGeneratedColumn } from 'typeorm'
+import { Profile } from './profile'
+import { Post } from './post'
+
+@Entity('post_like')
+export class PostLike extends BaseEntity {
+    @PrimaryGeneratedColumn()
+    id: number;
```
Comment (AjobK) [inline backend/app/entity/post_like.ts]: semicolon

File: backend/app/service/postService.ts [hunk 13]
```diff
@@ -2,10 +2,12 @@ import { Request, Response } from 'express'
 import PostDAO from '../dao/postDao'
 import Post from '../entity/post'
 import { v4 as uuidv4 } from 'uuid'
-import ProfileDAO from '../dao/profileDao'
+import ProfileDAO from '../dao/ProfileDao'
 import AccountDAO from '../dao/accountDao'
 const jwt = require('jsonwebtoken')
 require('dotenv').config()
+import PostLike from '../entity/post_like'
+import Profile from '../entity/profile';
```
Comment (AjobK) [inline backend/app/service/postService.ts]: Semi

**Review by AjobK — CHANGES_REQUESTED** [review]
Semi colons

**Pull request merged**